### PR TITLE
Changes to torch and delayer appearance 

### PR DESCRIPTION
**Pull request by mruncreative**
Changed torch to be plantlike, x-shaped instead of just a sprite and only use 1 texture instead of 3 and changed delayer to require only 3 textures from texturepacks instead of 13 changing the model to be a simple slab. Yes, I'm sad for whoever put in the work into the delayer model but it clashes with texturepacks and requires way to much work. Especially if they are not 16x16.

### CODE REVIEW

**Review by Niklp09 — COMMENTED** [review]
Just some comments from my side:
- The new delayer textures look awful IMHO
- Sharing some textures between luacontroller and microcontroller seems fine
- Sharing piston textures seems fine too
- The new torch style looks very nice

screenshot_20240420_091742

**Comment by mruncreative** [comment]
Do you have any ideas for the delayer? The original delayer doesn't look that great either and is super hard to texture. This one just needs two textures from texture pack creators.
I could make it look like the gates. Maybe that would fit in better? But then it would require the gates or need more textures.

**Comment by mruncreative** [comment]
grafik
Would you approve of this? Since gates depend on the delayer anyway, I thought I could move their textures to the delayer mod (without renaming them) and make the delayer look like a gate too.
Haven't applied the changes to the branch yet though.
Actually, why do the gates depend on the delayer? I don't see any reason for them to. Should I move the gates textures to the base mesecons mod and remove the delayer dependency of the gates mod, so things stay modular? But that would bloat the base.

**Comment by mruncreative** [comment]
Before anyone asks: License of my arrow of slowness and code changes is CC0

**Comment by Niklp09** [comment]
> License of my arrow of slowness and code changes is CC0

Not sure how the license situation here is, acording to the [readme](https://github.com/minetest-mods/mesecons?tab=readme-ov-file#alright-how-can-i-use-it) all media files are cc-by-sa 3.0.

**Comment by mruncreative** [comment]
I said CC0 meaning I don't give a shit what you do with it or how you relicense it.

File: mesecons_gates/mod.conf
Comment (Niklp09) [inline mesecons_gates/mod.conf:1]: reason for removing `mesecons_delayer` from the deps?

**Comment by Niklp09** [comment]
> I would like to know what you think of the new delayer and what still needs to be done

I like the "old" delayer design as it looks (more) like an arrow, this is pretty subjective.

File: mesecons_gates/mod.conf
Comment (mruncreative) [reply mesecons_gates/mod.conf:1]: No code in gates seems to require the delayer. Not even the crafting recipes.

**Comment by mruncreative** [comment]
If no one cares anyway, why not just merge?

**Comment by Niklp09** [comment]
> why not just merge?

Because no one approved your changes, I recommend to move the uncontroversial stuff to a second PR.

> If no one cares anyway

Probably due to your previous behavior...

**Comment by mruncreative** [comment]
This PR has been up for a month now. I asked if anybody would like to volunteer reviewing this on IRC two days ago but nobody even replied.
https://irc.minetest.net/minetest/2024-05-10

I really think the people who are active don't care and those who aren't active I can't ask either way. I don't see how it's a controversial change. I doesn't break any other texture packs or mods to my knowledge except for the top of delayer in existing texture packs. However the old delayer is an ugly piece of shit, that requires 13 texture that are variations and the elevated voxels of the circle assume a 16x16 texture pack and look shit with all other resolutions like 32x32, 64x64, 128x128. Here is a picture from mesecons.net. I think it's the hdx-128 texture pack. It's objectively ugly. The other mesecons textures in the pack look better. I think the creator of the pack gave up on it and this monstrosity was born.
grafik

**Review by SmallJoker — COMMENTED** [review]
Things I like:
1. rotated piston textures. Reduces texture count, which is nice.
2. the torch looks much better now

Things I don't mind:
1. Delayer nodebox. Unified design with gates. Fine by me. The old design looked good and a simpler version is not bad either.
2. Delayer texture. Very simplistic, yet fits the other gates, thus OK.
   * EDIT: I'd welcome the design shown above with the straight arrow. I this this preserves the legacy better while still keeping the functionality of the gate visible (dashed arrow).


Things where I don't agree:
1. `jeija_gate_*.png` should be kept in the `mesecons_gates` mod. These texture are too specific for the base mod. Perhaps make the delayer depend on gates?
4. `jeija_luacontroller_*.png` are too specific for `mesecons`.  But what would be a better place to put them?
5. Delayer appearance: the output side is unlike the other gates. It should indicate whether the output is active (currently not visible).

File: mesecons_delayer/init.lua [hunk 1]
```diff
@@ -74,16 +63,16 @@ end
...
+	inventory_image = "jeija_gate_off.png^jeija_delayer.png",
+	wield_image = "jeija_gate_off.png^jeija_delayer.png",
 	tiles = {
-		"mesecons_delayer_off_"..tostring(i)..".png",
-		"mesecons_delayer_bottom.png",
-		"mesecons_delayer_ends_off.png",
-		"mesecons_delayer_ends_off.png",
-		"mesecons_delayer_sides_off.png",
-		"mesecons_delayer_sides_off.png"
+		"jeija_microcontroller_bottom.png^jeija_gate_output_off.png^jeija_gate_off.png^jeija_delayer.png^mesecons_delayer_"..tostring(i)..".png",
```
Comment (SmallJoker) [inline mesecons_delayer/init.lua:69]: Would you please be so nice to split this line? 138 characters is pretty long. Same for L112 below.

**Review by SmallJoker — APPROVED** [review]
comparison

> the jeija_microcontroller_LED_*.png were renamed to jeija_luacontroller_LED_*.png because digistuff depends on the luacontroller leds and I don't want to break it.

(partially a note to myself) This must be put into the final commit  message to preserve the reason for this renaming.

Aside from the delayer, the other texture changes lead to no visible change with only the "base" texture pack being active, which is good. In the future, more contrast on the gate/delayer textures could make them easier to look at.

Fine by me. If there's any objections, please let me know. If there are none, I'll go ahead and merge this PR in a week.

File: mesecons_delayer/init.lua (hunk 1 above)
Comment (mruncreative) [reply mesecons_delayer/init.lua:69]: Get a better text editor.

**Review by Niklp09 — APPROVED** [review]
LGTM. Thanks for improving the overall consistency of mesecons.

**Comment by mruncreative** [comment]
**TIME TO MERGE ☺**

**Pull request merged**